### PR TITLE
Feature/multiple targets build

### PR DESCRIPTION
**Pull request by jabou**
This PR adds support for selecting multiple configurations at the same time.

https://github.com/infinum/app-deploy-script/assets/11990539/db0339ac-fa97-4d5d-bc63-9fa11845d3de

### CODE REVIEW

File: deploy-options.sh [hunk 1]
```diff
@@ -40,37 +40,21 @@ function deploy_options {
...
+    echo "[2] Production"
     echo
     echo "=================="
     echo
     echo "---------------------"
     echo "| APP STORE CONNECT |"
     echo "---------------------"
     echo
-    echo "[4] App Store"
+    echo "[3] App Store"
```
Comment (hhrvoic) [inline deploy-options.sh:53]: Can we make this dynamic? We define possible envs somewhere at the start of the script, for example as dictionary where key is environment's id (e.g. `"internal-staging"` and value is the display name, e.g. `"Staging"`). This way we don't have to fix prints every time we fiddle with the array of options.

File: app-deploy.sh [hunk 2]
```diff
@@ -108,6 +109,23 @@ function initial_checkup {
...
+
+    # Parse all selected options
+    IFS=', ' read -r -a environments_array <<< "$target_selection"
+
+    environments_to_build=()
+
+    for environment in "${environments_array[@]}"; do
+        if [ ${environment} -le 9 -a ${environment} -ge 0 ]; then
+            environments_to_build+=("${environments[${environment}]}")
+        else
```
Comment (hhrvoic) [inline app-deploy.sh]: Why do we check 0-9 range, I mean, what if there is 10+ envs? Not common for most of projects, but could happen. Should we just check if the environment number is not larger of the possible env count? Sry if I misunderstood the logic.

**Review by hhrvoic — APPROVED** [review]
Looks good, added small improvement suggestions

File: deploy-options.sh (hunk 1 above)
Comment (jabou) [reply deploy-options.sh:53]: I would like to keep this one as-is. The user should define this part of the script so they can write whatever text/format they want here. 

And this is only an example/template that they have to change.

File: app-deploy.sh (hunk 2 above)
Comment (jabou) [reply app-deploy.sh]: Oh, you are right! 

I totally missed that one. I'll try to extract a number of items in the array and then check between 0 and the array.count.
If Bash doesn't have this, I'll remove this check entirely.

Comment (hhrvoic) [reply app-deploy.sh]: Yeah, I think we can use`${#arrayName[@]}` for that :)

**Review by nikolamajcen — APPROVED** [review]

File: app-deploy.sh (hunk 2 above)
Comment (jabou) [reply app-deploy.sh]: I had to add `$((${#environments[@]} - 1))` on that place to get max - 1 (counting from 0).

**Pull request merged**